File: data structures and algorithms/java/05_035.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
import java.math.*;

public class Main {
    public static void main(String[] args) {
        // Main method code goes here
    }
    
    public class Graph {
        private Map<Object, List<Object>> graph;
        
        public Graph() {
            this.graph = new HashMap<>();
        }
        
        public void addVertex(Object vertex) {
            if (!this.graph.containsKey(vertex)) {
                this.graph.put(vertex, new ArrayList<>());
            }
        }
        
        public void addEdge(Object v1, Object v2) {
            if (!this.graph.containsKey(v1)) {
                this.addVertex(v1);
            }
            if (!this.graph.containsKey(v2)) {
                this.addVertex(v2);
            }
            this.graph.get(v1).add(v2);
        }
    }
}

File: data structures and algorithms/cs/05_035.cs
using System;
using System.Collections.Generic;

public class Graph
{
   private Dictionary<object, List<object>> graph;

   public Graph()
   {
      this.graph = new Dictionary<object, List<object>>();
   }

   public void AddVertex(object vertex)
   {
      if (!this.graph.ContainsKey(vertex))
      {
         this.graph[vertex] = new List<object>();
      }
   }

   public void AddEdge(object v1, object v2)
   {
      if (!this.graph.ContainsKey(v1))
      {
         AddVertex(v1);
      }
      if (!this.graph.ContainsKey(v2))
      {
         AddVertex(v2);
      }
      this.graph[v1].Add(v2);
   }

   public IEnumerable<object> GetAllVertices()
   {
      return graph.Keys;
   }

   public bool HasEdge(object v1, object v2)
   {
      return graph.ContainsKey(v1) && graph[v1].Contains(v2);
   }
}
